### PR TITLE
add progress bar

### PR DESCRIPTION
**Pull request by madhur-tandon**
Closes https://github.com/jupyterlab-contrib/search-replace/issues/23

@fcollonval not sure I can verify if this works -- since the query is too fast I guess OR I might be doing it incorrectly

### CODE REVIEW

**Comment by fcollonval** [comment]
You could try wrapping in a Promise.all to waitforselector (progessbar) and fill search at the same time. Hopefully the progress bar is at least displayed in one frame.

**Comment by madhur-tandon** [comment]
@fcollonval I can see it, though it's available for a very short time. The fix was to emit state change in it's setter.

File: src/searchReplace.tsx [hunk 1]
```diff
@@ -44,13 +55,15 @@ export class SearchReplaceModel extends VDomModel {
       this._queryResults = data;
       this.stateChanged.emit();
       console.log(data);
+      this.isLoading = false;
```
Comment (fcollonval) [inline src/searchReplace.tsx:58]: We should move this after the try-catch block to ensure it is turn off even if an error occurs.

File: src/searchReplace.tsx [hunk 2]
```diff
@@ -44,13 +55,15 @@ export class SearchReplaceModel extends VDomModel {
       this._queryResults = data;
       this.stateChanged.emit();
       console.log(data);
```
Comment (fcollonval) [inline src/searchReplace.tsx:57]: Let's remove that debug print

```suggestion
```

File: src/searchReplace.tsx [hunk 3]
```diff
@@ -2,17 +2,27 @@ import React from 'react';
...
     });
   }
 
+  get isLoading(): boolean {
+    return this._isLoading;
+  }
+
+  set isLoading(v: boolean) {
+    this._isLoading = v;
+    this.stateChanged.emit();
```
Comment (fcollonval) [inline src/searchReplace.tsx:23]: Let's emit the signal only if `v !== this._isLoading`.

File: src/searchReplace.tsx [hunk 4]
```diff
@@ -74,6 +87,7 @@ export class SearchReplaceView extends VDomRenderer<SearchReplaceModel> {
             (this.model.searchString = event.target.value)
           }
         />
+        {this.model.isLoading && <Progress />}
```
Comment (fcollonval) [inline src/searchReplace.tsx:90]: Could you switch to `ProgressRing` to integrate better with JLab pattern?

And could you add style to place the progress ring in the center of the panel?

File: src/searchReplace.tsx (hunk 1 above)
Comment (madhur-tandon) [reply src/searchReplace.tsx:66]: done

File: src/searchReplace.tsx (hunk 2 above)
Comment (madhur-tandon) [reply src/searchReplace.tsx:65]: done

File: src/searchReplace.tsx (hunk 3 above)
Comment (madhur-tandon) [reply src/searchReplace.tsx:31]: done

**Pull request merged**